lib/functions/ui.functions.sh

```diff
@@ -37,6 +45,10 @@ toggle_spin() {
                 done
             ) &
             spin_pid=$!
+            process_trap_add INT _ui_spinner_kill
+            if ! process_trap_get_exit INT >/dev/null; then
```

Comment:
After looking at this a little longer, it feels a bit unwieldy.

Maybe we could change `process_trap_set_exit` s.t. it only sets the exit code supplied if one isn't already set?

Perhaps:

```
process_trap_exit_on INT
```
...With a default of exit code `1` if it's not supplied?